### ISSUE TITLE
`headOption` method does not compile for some `Array[Nothing]`

### ISSUE THREAD

**Issue by Atry** [scala/bug#10590]
```
Welcome to Scala 2.12.4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_131).
Type in expressions for evaluation. Or try :help.

scala> Array.empty.headOption // OK
res0: Option[Nothing] = None

scala> val emptyArray = Array.empty
emptyArray: Array[Nothing] = Array()

scala> emptyArray.headOption // Does not compile
<console>:13: error: value headOption is not a member of Array[Nothing]
       emptyArray.headOption // Does not compile
                  ^
```

**Comment by Jasper-M** [comment]
Looks like a classic case of the `Nothing` bug. Can't immediately find the most relevant issue though.

**Comment by som-snytt** [comment]
https://github.com/scala/bug/issues/9453

**Comment by xuwei-k** [comment]
in Scala-2.13.0-M4 `Array.empty[Nothing].headOption` compile success. but it call low priority [low priority `copyArrayToImmutableIndexedSeq`](https://github.com/scala/scala/blob/v2.13.0-M4/src/library/scala/Predef.scala#L810) not [`genericArrayOps`](https://github.com/scala/scala/blob/v2.13.0-M4/src/library/scala/Predef.scala#L436). 😕

```
Welcome to Scala 2.13.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> Array.empty[Nothing].headOption
                  ^
       warning: method copyArrayToImmutableIndexedSeq in class LowPriorityImplicits2 is deprecated (since 2.13.0): Implicit conversions from Array to immutable.IndexedSeq are implemented by copying; Use the more efficient non-copying ArraySeq.unsafeWrapArray or an explicit toIndexedSeq call
res0: Option[Nothing] = None
``` 

another example

```scala
object Test {
  implicit class A[B](value: Array[B]) {
    def x: Int = 42
  }

  implicit class C[D](value: List[D]) {
    def y: Int = 42
  }

  // fail
  Array.empty[Nothing].x

  // success
  Array.empty[String].x

  // success
  List.empty[Nothing].y

  // success
  List.empty[String].y
}
```

**Comment by szeiger** [comment]
As mentioned above, this specific instance of the bug is fixed in 2.13, so I'm closing this ticket. The more general bug is https://github.com/scala/bug/issues/9453, which appears to be fixed in Dotty.
